File: client-scalaz/src/main/scala/org/openapitools/client/api/TicketListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import TicketListResponse._

case class TicketListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[TicketResponse]],
count: Option[Integer],
hasMoreResults: Option[Boolean],
start: Option[Integer],
limit: Option[Integer],
countTotal: Option[Long],
returning: Option[String])

object TicketListResponse {
  import DateTimeCodecs._

  implicit val TicketListResponseCodecJson: CodecJson[TicketListResponse] = CodecJson.derive[TicketListResponse]
  implicit val TicketListResponseDecoder: EntityDecoder[TicketListResponse] = jsonOf[TicketListResponse]
  implicit val TicketListResponseEncoder: EntityEncoder[TicketListResponse] = jsonEncoderOf[TicketListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/ticket_list_response.clj
(ns sirqul-io-t-platform.specs.ticket-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.ticket-response :refer :all]
            )
  (:import (java.io File)))


(def ticket-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of ticket-response-spec)
   (ds/opt :count) int?
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :start) int?
   (ds/opt :limit) int?
   (ds/opt :countTotal) int?
   (ds/opt :returning) string?
   })

(def ticket-list-response-spec
  (ds/spec
    {:name ::ticket-list-response
     :spec ticket-list-response-data}))
